changelog/unreleased/pull-3776

```diff
@@ -0,0 +1,8 @@
+Bugfix: Fix handling of `maxKeys` in `SearchKey`
+
+When there are many key files in the repository and wrong passwords are
+entered, the process of matching passwords is very slow. Now restic can
+retry up to 20 times, or you can use `-key-hint=<Key ID>` to specify
```

Comment:
This currently sounds like an increase, whereas in fact it decreases the number of key files tested. The previous state (since `maxKeys` was added to the code in 2016) was to test the password against every key in the repository.